python/ray/data/_internal/execution/streaming_executor_state.py

```diff
@@ -513,6 +513,12 @@ def process_completed_tasks(
         for ref in ready:
             state, task = active_tasks[ref]
             ready_tasks_by_op[state].append(task)
+        for ref in remaining:
+            state, task = active_tasks[ref]
+            # If there are pending blocks, try to process them even if streaming_gen is not ready
+            # This may be because the (block, meta) pair processing encountered exceptions or timeouts, leaving data unconsumed
+            if isinstance(task, DataOpTask) and task.pending_block_pair:
+                ready_tasks_by_op[state].append(task)
```

Comment:
In the case where a map task hasn't finished reconstructing the metadata, what's the disadvantage of waiting for the metadata to be ready before adding it to `ready_tasks_by_op` (i.e., what happens if we don't include these lines)?